sql/parser/sql.y

```diff
@@ -3990,6 +4017,7 @@ type_func_name_keyword:
 | CURRENT_SCHEMA
 | FREEZE
 | FULL
+| GRANTS
```

Comment:
I imagine the parser conflicts are due to the same problem that `SHOW {DATABASES,TABLES}` has and can probably be solved in a similar manner. That is, detection of `SHOW GRANTS` in the `SHOW var_name` production.